testnet/stacks-node/src/neon_node.rs

```diff
@@ -1680,6 +1738,52 @@ impl InitializedNeonNode {
             Some(event_observer),
         ) {
             Ok(block) => block,
+            Err(ChainstateError::InvalidStacksMicroblock(msg, mblock_header_hash)) => {
+                // part of the parent microblock stream is invalid, so try again
+                info!("Parent microblock stream is invalid; trying again without the offender {} (msg: {})", &mblock_header_hash, &msg);
+
+                // truncate the stream
+                stacks_parent_header.microblock_tail = match microblock_info_opt {
+                    Some((microblocks, _)) => {
+                        let mut tail = None;
+                        for mblock in microblocks.into_iter() {
```

Comment:
Maybe the iteration logic here could be avoided by modifying `process_microblocks_transactions` (which seems is like what is ultimately returning the data for the `InvalidStacksMicroblock` error here) to return an option with the previous microblock hash?

Reply:
As a general rule, I try to write methods that only return data the caller can't figure out on its own; otherwise, the method could be doing excess work that the caller doesn't need.